lib/cloudregister/registerutils.py

```diff
@@ -159,10 +159,13 @@ def clear_rmt_as_scc_proxy_flag():
 def credentials_files_are_equal(repo_credentials):
     """Compare the base credentials files the the repo header and make
        sure they have the same values."""
-    credentials_location = '/etc/zypp/credentials.d/'
+
     if not repo_credentials or not isinstance(repo_credentials, str):
         return False
 
+    base_credentials_location = '/etc/zypp/credentials.d/'
+    target_root = get_zypper_target_root()
+    credentials_location = target_root + base_credentials_location
```

Comment:
Would it be worth join the path fragments with `os.path.join`  to avoid possible '//' when adding the strings?   (I assume `os.path.userexpand` is not necessary for the paths we're processing)

Reply:
The path we get back from reading the process table is formed by the transactional-update command and does not end in a `/`. Also `os.path.join` has some weird rules and to make that work `get_zypper_target_root()` would need to return `\` instead of an empty string and `base_credentials_location` would have to start without a `/`. Meaning one has to be rather careful with the construction of the arguments passed to `os.path.join` . Besides a `//` does not hurt.

Since we will operate under the conditions of transactional-update I think we are safe also skipping `userexpand`. 

There are use cases where the underlying assumptions of operating in an environment constructed `transactional-update` will be violated and the code will fail. IF ever needed we'll come back and look at that specific use case.